python/lsst/analysis/tools/actions/vector/selectors.py

```diff
@@ -167,6 +168,17 @@ def setDefaults(self):
         self.selectWhenTrue = ["detect_isPatchInner", "detect_isDeblendedSource"]
 
 
+class MatchingFlagSelector(CoaddPlotFlagSelector):
+    """The default flag selector to apply pre matching.
+    The sources are cut down to remove duplicates but
+    not on quality.
+    """
+
+    def setDefaults(self):
+        self.selectWhenFalse = []
+        self.selectWhenTrue = ["detect_isPrimary"]
```

Comment:
Shouldn't this have `detect_isTractInner` for coadds?

Reply:
Not when working per-tract as there is no chance of duplication if there's no tract overlap.  Setting it in a per-tract context will unnecessarily leave an empty boundary around the tract (and will almost certainly puzzle people as to why!)  If plotting data with overlapping tracts, then it is best just to use `isPrimary`.

Reply:
I thought isPrimary included isTractInner?

Reply:
I rather meant to not change the selection in this ticket (except for excluding sky objects, perhaps) as it will be difficult to identify the change which is unrelated to the ticket when we are trying to figure out why the metrics changed…